modules/stochastic_tools/src/transfers/SamplerPostprocessorTransfer.C

```diff
@@ -0,0 +1,84 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+// StochasticTools includes
+#include "SamplerPostprocessorTransfer.h"
+#include "SamplerMultiApp.h"
+#include "SamplerReceiver.h"
+#include "StochasticResults.h"
+
+template <>
+InputParameters
+validParams<SamplerPostprocessorTransfer>()
+{
+  InputParameters params = validParams<MultiAppTransfer>();
+  params.addClassDescription("Transfers data to and from Postprocessors on the sub-application.");
+  params.addParam<VectorPostprocessorName>(
+      "results",
+      "Name of the StochasticResults object that the results are to be stored or extracted.");
+  params.addParam<std::string>(
+      "postprocessor", "Name of the Postprocessor on the sub-application to transfer to or from.");
+  params.set<MooseEnum>("direction") = "from_multiapp";
+  params.suppressParameter<MooseEnum>("direction");
+  return params;
+}
+
+SamplerPostprocessorTransfer::SamplerPostprocessorTransfer(const InputParameters & parameters)
+  : MultiAppTransfer(parameters),
+    _results_name(getParam<VectorPostprocessorName>("results")),
+    _sampler_multi_app(std::dynamic_pointer_cast<SamplerMultiApp>(_multi_app).get()),
+    _sampler(_sampler_multi_app->getSampler()),
+    _sub_pp_name(getParam<std::string>("postprocessor"))
+{
+  if (!_sampler_multi_app)
+    mooseError("The 'multi_app' must be a 'SamplerMultiApp.'");
+}
+
+void
+SamplerPostprocessorTransfer::initialSetup()
+{
+  const ExecuteMooseObjectWarehouse<UserObject> & user_objects = _fe_problem.getUserObjects();
+  std::shared_ptr<UserObject> uo = user_objects.getActiveObject(_results_name);
+  _results = std::dynamic_pointer_cast<StochasticResults>(uo).get();
```

Comment:
Minor: This line of code is weird but OK. The `dynamic_pointer_cast` might fail but then you are calling `get` on it anyway. I checked the standard and this is OK, but it's a little odd. Since you are just throwing a MOOSE error if it fails, you could probably restructure this just a bit to clean it up.